### ISSUE TITLE
Explore use of Portable Software Instance (PSI)

### ISSUE THREAD

**Issue by armstro** [zowe/zlc#168]
Proposal to support Zowe install by Portable Software Instance(PSI) - research needed - software prereqs, use of build environment, automation, etc.....

**Comment by armstro** [comment]
Hi Sujay and Bruce,


let me summarize the idea technically. Currently, there are two versions of Zowe distribution, convenience build, and SMP/E build. During the packaging of Zowe, SMP/E environment is created. This environment is the starting point for the PSI creation process. SMP/E location, a few additional variables, and z/OSMF workflows are fed to the z/OSMF using REST API. As a result of this call, z/OSMF generates a JCL. The JCL can be submitted and the result of the job is a PSI. I believe (and here I could be wrong) that the 2.3 z/OMSF version is sufficient for the PSI creation (please ask Marna to confirm that). We have now two approaches ready, the one with the REXX script and the other with the shell script. As far as I know, (and Jack can expand on this, as he is working with my team), the preferred way is to use the shell script, which will be part of the Jenkins CI/CD. What could be confusing is that we are working on it in the Cupids squad, not in the CI/CD squad.
So, to sum it up, we know how to do it, we have people to do it, I believe we have the environment ready as well. The only thing missing now is the agreement to do that.


Best Regards
Jan

**Comment by Joe-Winchester** [comment]
The Cupids squad has a 30 minute scrum three times a week https://lists.openmainframeproject.org/g/zowe-dev/calendar - tomorrow Friday March 6th it is at 10am EST.  Would it make sense for us to chat about this then, or else the architecture call every week is Tuesday 9am EST https://lists.openmainframeproject.org/g/zowe-dev/viewevent?repeatid=21909&eventid=683269&calstart=2020-04-14 which is 1 hour long and this could be a good discussion to have with the wider squads who attend to work out any remaining issues that are holding this up from being completed ?

**Comment by jp669844** [comment]
I have a day off tomorrow, so if you need more details from me, I would prefer to talk about it on Tuesday at the architecture call.

**Comment by Joe-Winchester** [comment]
@nkocsis and @1000TurquoisePogs - could we please add Jan to the agenda of next Tuesday's architecture call.  I'll also try to invite someone from IBM who can present on PSI and help set the stage.
